ui/webui/src/components/installation/InstallationProgress.jsx

```diff
@@ -96,7 +98,7 @@ export class InstallationProgress extends React.Component {
                     </EmptyStateBody>
                     {status === "success" &&
                     <EmptyStateSecondaryActions>
-                        <Button>{_("Reboot")}</Button>
+                        <Button onClick={exitGui}>{_("Reboot")}</Button>
```

Comment:
Hmm, I was actually expecting to do a real reboot. Aka, cockpit.spawn("reboot -f")

How come killing the gui process does a reboot?

Reply:
Reboot is the final effect, but before that, Anaconda needs to also run some other stuff like reporting IPMI status etc. That is the proper code path to use, regardless of how we want to call it.

As to why it works - check `CockpitUserInterface.run()`. The Anaconda UI process waits until `webui-desktop` is finished. Killing the script lets the process continue, which means it just runs to the end and exits. And that is when `exitHandler` runs and does the cleanup and actual reboot.

Reply:
That isn't really possible, I think. Once the script process is killed, anaconda goes through termination, shuts down system with systemctl, and that then probably kills the rest of the processes. The point is that there might be some delay, but it's not clear how much, and it's also not clear if it's possible to run more processes after that happens.